exetera/core/fields.py

```diff
@@ -340,6 +365,8 @@ def write(self, part):
         :param part: numpy array to write to field
         :return: None
         """
+        if len(self._references) > 0:
+            self.concreate_all_fields()
```

Comment:
move these into field, so that the callback flow is: fieldarray -> field -> field_references